ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/CounterNamingStrategy.java

```diff
@@ -0,0 +1,358 @@
+ *   <li>Convert all elements within each voltage level
+ *   <li>Convert network-level elements
+ * </ol>
+ *
+ * Each element receives a unique sequential identifier during the conversion process.
+ *
+ * @param network the network to convert
+ */
+    @Override
+    public void initialiseNetwork(Network network) {
+        namingCounter = 0;
+        network.getSubstations().forEach(substation -> substation.getVoltageLevels().forEach(voltageLevel -> {
+
+            System.out.println(voltageLevel.getId());
+            voltageLevel.getBusBreakerView().getBuses().forEach(bus -> {
+                generateUcteNodeId(bus.getId(), voltageLevel);
+                logConversion("BUS", bus);
+            });
+            voltageLevel.getBusBreakerView().getSwitches().forEach(sw -> {
+                if (sw.getKind().equals(SwitchKind.BREAKER)) {
+                    generateUcteElementId(sw);
+                } else {
+                    generateUcteNodeId(sw.getId(), voltageLevel);
+                }
+                logConversion("SW", sw);
+            });
+        }));
+
+        network.getBranchStream().forEach(branch -> {
+            generateUcteElementId(branch);
+            logConversion(branch.getType().name(), branch);
+        });
+
+        network.getDanglingLineStream().forEach(d -> {
+            generateUcteNodeId(d);
+            generateUcteElementId(d);
+            logConversion(d.getType().name(), d);
+        });
+
+    }
+
+    private void logConversion(String elementType, Identifiable element) {
+        System.out.println(String.format("%s : %s--> %s %s",
+                elementType,
+                element.getId(),
+                ucteNodeIds.get(element.getId()),
+                ucteElementIds.get(element.getId())
+        ));
+
+    }
+
+    /**
+     * Generates a UCTE node code for a network element.
+     * The generation process involves:
+     * <ol>
+     *   <li>Checking for existing node ID to avoid duplicates</li>
+     *   <li>Extracting country code from the voltage level</li>
+     *   <li>Determining voltage level code based on nominal voltage</li>
+     *   <li>Creating a new node ID using namingCounter</li>
+     * </ol>
+     *
+     * The generated UCTE node code follows the format: [CountryCode][NodeId][VoltageLevelCode]
+     *
+     * @param nodeId original identifier of the network element
+     * @param voltageLevel voltage level containing the element
+     * @return generated UcteNodeCode, or null if nodeId already exists in ucteNodeIds
+     */
+    private UcteNodeCode generateUcteNodeId(String nodeId, VoltageLevel voltageLevel) {
+
+        if (ucteNodeIds.containsKey(nodeId)) {
+            return ucteNodeIds.get(nodeId);
+        }
+        if (namingCounter > 9999) {
+            namingCounter = 0;
+        }
+
+        StringBuilder newNodeCode = new StringBuilder(8);
+        String newNodeId = String.format("%05d", ++namingCounter);
```

Comment:
The check is against 4 digits, but the `String.format` pattern uses 5 digits.
Couldn't the condition at line 117 be `if (namingCounter > 99_999)` (5 digits instead of 4)?

Another remark about this part: if `namingCounter` is reset to 0, it may produce an invalid file (several nodes with the same id). I think throwing an exception would be safer.

Reply:
i create a new function "generateIDFromCounter" for generate nodeID